OpenRA.Mods.Common/Widgets/WorldInteractionControllerWidget.cs

```diff
@@ -245,22 +249,24 @@ public override string GetCursor(int2 screenPos)
 
 		public override bool HandleKeyPress(KeyInput e)
 		{
-			var player = World.RenderPlayer ?? World.LocalPlayer;
+			// Players to be included in the selection (the viewer or all players in "Disable shroud" / "All players" mode)
+			var viewer = (World.LocalPlayer == null || World.LocalPlayer.Spectating) ? World.RenderPlayer : World.LocalPlayer;
+			var eligiblePlayers = viewer == null || (viewer.NonCombatant && viewer.Spectating) ? World.Players : World.Players.Where(p => p == viewer);
```

Comment:
Isn't `World.Players.Where(p => p == viewer)` just a confusing way of saying `new [] { viewer }`?

Reply:
Yes, pardon me, I didn't know I could do that :-) still learining the language. Fixed.